### PR TITLE
Search Improvements: show podcasts and episodes in a standalone screen

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Project: #709 |
|:---:|

Show podcasts and episodes on a standalone screen when tapping "Show All".

In this PR I also changed the max number of episodes to show in the search results view to 20 to match Android.

### Before everything

1. Run the app
2. Go to Profile > Settings > Beta Features > enable `newSearch`
3. Change the app Build Configuration to `Staging`

### Showing all podcasts

1. Go to discover
2. Perform a search
3. Tap "Show All" for podcasts
4. ✅ A screen containing all the podcasts from the results should appear
5. Tap any podcast
6. ✅ The podcast screen should appear
6. Go back to the search history screen
6. The podcast you just opened should be there

### Showing all podcasts

1. Go to discover
2. Perform a search
3. ✅ A max of 20 episodes should be displayed in the search main screen
3. Tap "Show All" for episodes
3. ✅ All episodes should be there
3. Tap any episode
3. ✅ The episode screen should appear
6. Go back to the search history screen
6. The episode you just opened should be there

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/New Search/Views/Results/InlineResultsView.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+import PocketCastsServer
+
+struct InlineResultsView: View {
```
Comment (leandroalonso) [inline podcasts/New Search/Views/Results/InlineResultsView.swift]: I'm not sure this name makes sense, open to suggestions.

File: podcasts/New Search/Views/Results/InlineResultsView.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @EnvironmentObject var theme: Theme
+
+    @ObservedObject var searchResults: SearchResultsModel
+
+    let searchHistory: SearchHistoryModel?
+
+    /// If this view should show podcasts or episodes
+    var showPodcasts = true
```
Comment (emilylaguna) [inline podcasts/New Search/Views/Results/InlineResultsView.swift]: What do you think about making this an enum and renaming it to `displayMode`? 

I think doing that would provide some extra context without needing to read the doc comment.

File: podcasts/New Search/Views/Results/InlineResultsView.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                        } else {
+                            ForEach(searchResults.episodes, id: \.self) { episode in
+
+                                SearchResultCell(episode: episode, podcast: nil, searchHistory: searchHistory)
+                            }
+                        }
+                    }
+                }
+            }
+            .navigationBarTitle(Text(showPodcasts ? L10n.discoverAllPodcasts : "All Episodes"))
```
Comment (emilylaguna) [inline podcasts/New Search/Views/Results/InlineResultsView.swift]:
```suggestion
            .navigationBarTitle(Text(showPodcasts ? L10n.discoverAllPodcasts : L10n.discoverAllEpisodes))
```

File: podcasts/New Search/Views/Results/InlineResultsView.swift (hunk 1 above)
Comment (emilylaguna) [reply podcasts/New Search/Views/Results/InlineResultsView.swift]: `SearchResultsListView`?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Working as intended, left a couple comments below.

File: podcasts/New Search/Views/Results/InlineResultsView.swift (hunk 2 above)
Comment (leandroalonso) [reply podcasts/New Search/Views/Results/InlineResultsView.swift]: Done in c556e99d677ea4777f9217f531bf28c8f3ba4225

File: podcasts/New Search/Views/Results/InlineResultsView.swift (hunk 1 above)
Comment (leandroalonso) [reply podcasts/New Search/Views/Results/InlineResultsView.swift]: Yes! Done in 098ea0d148bfdb9c81ac99ddeb0c86bee02599bf

**Review by emilylaguna — APPROVED** [review]
🚢

**Pull request merged**